csrc/fusion_profiler.h

```diff
@@ -236,29 +237,31 @@ class SegmentProfiler {
 //! \brief A singleton class to profile Fusions that can include multiple
 //! segments.
 class FusionProfiler {
+ private:
   FusionProfiler();
+  ~FusionProfiler();
 
-  //! Method to access FusionProfiler singleton
-  static FusionProfiler* get();
+  FusionProfiler(const FusionProfiler&) = delete;
+  FusionProfiler& operator=(const FusionProfiler&) = delete;
 
  public:
-  NVF_API static void reset();
```

Comment:
Why is it okay to remove the NVF_API calls?  I thought we needed these for usage exposure outside of the class.

Reply:
correct me if I'm wrong, I think we would need these if we are eve to expose them via python API?! Is that on the map somewhere tagging @rdspring1

Reply:
We don't have to resolve them here. It's easy to add back if needed later.

Reply:
Yes, but we can add it back on a need basis.
I foresee only one or two methods around tearing down CUPTI being exposed to Python. Hence, it is still unnecessary for other methods.